src/detection/Peak.cc

```diff
@@ -185,9 +141,7 @@ PTR(PeakTable) PeakTable::make(
         }
     }
     // No match: we create a new table and put it in the cache
-    PTR(PeakTable) newTable = std::make_shared<PeakTableImpl>(
-        schema, afw::table::IdFactory::makeSimple()
-    );
+    std::shared_ptr<PeakTable> newTable(new PeakTable(schema, afw::table::IdFactory::makeSimple()));
```

Comment:
Sorry, can you clarify *what* is being friended? I don't see how the use of `make_shared` puts any restrictions on something like that.